### PR TITLE
basic auth login UI

### PR DESCRIPTION
**Pull request by kunmingg**
Frontend part.

Test passed, ready to merge.

related: https://github.com/kubeflow/kubeflow/pull/2262

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2316)

### CODE REVIEW

File: components/kflogin/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+## Basic Auth Login Page
+
+UI to login kubeflow cluster.
```
Comment (jlewi) [inline components/kflogin/README.md]: UI to login into Kubeflow cluster using basic auth

**Comment by jlewi** [comment]
/lgtm
/approve
/hold

@yebrahim Would you mind taking a look at this?

File: components/kflogin/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+The page will reload if you make edits.<br>
+You will also see any lint errors in the console.
+
+### `npm test`
+
+Launches the test runner in the interactive watch mode.<br>
+See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+
+### `npm run build`
```
Comment (yebrahim) [inline components/kflogin/README.md]: Should we cleanup this readme? If there are no actual tests (or we're not planning to run them), let's remove the tests section, and we can change the build section below to mention using the makefile instead?

File: components/kflogin/src/serviceWorker.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
+// This optional code is used to register a service worker.
```
Comment (yebrahim) [inline components/kflogin/src/serviceWorker.js]: You generally won't need this file, it's safe to remove. So are its references in index.js.

File: components/kflogin/src/App.css [hunk 4]
```diff
@@ -0,0 +1,32 @@
+.App {
```
Comment (yebrahim) [inline components/kflogin/src/App.css]: This looks like the stock React CSS file, you probably aren't using anything here (.App is used in App.js but it's not doing anything). Can we remove it?

File: components/kflogin/src/App.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React, { Component } from 'react';
+import './App.css';
+import Login from './login';
+import { createBrowserHistory } from 'history';
+
+export const history = createBrowserHistory({
```
Comment (yebrahim) [inline components/kflogin/src/App.js]: This doesn't seem to be used anywhere.

File: components/kflogin/src/App.test.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (yebrahim) [inline components/kflogin/src/App.test.js]: I'm on the line about having this stock test, if we're never going to run it. I'm assuming any further developments here will make sure the app loads in the browser, which does more than what this file tests.

File: components/kflogin/src/index.css [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  margin: 0;
+  padding: 0;
+  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
+    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
+    sans-serif;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+}
+
+code {
```
Comment (yebrahim) [inline components/kflogin/src/index.css]: Is this used anywhere?

File: components/kflogin/src/App.css (hunk 4 above)
Comment (kunmingg) [reply components/kflogin/src/App.css]: We need ```text-align: center;```
Maybe leave this file for future style edit?

**Comment by kunmingg** [comment]
@yebrahim 
PR updated, ptal

**Comment by yebrahim** [comment]
/lgtm
Thanks.

**Comment by jlewi** [comment]
/cancel hold

Thanks @yebrahim

**Pull request merged**